hubspot/post.go

```diff
@@ -0,0 +1,38 @@
+package hubspot
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/spyzhov/ajson"
+)
+
+// post writes data to Hubspot. It handles retries and access token refreshes.
+func (c *Connector) post(ctx context.Context, url string, body any) (*ajson.Node, error) {
+	node, err := c.Client.Post(ctx, url, body)
+	if err != nil {
+		switch {
+		case errors.Is(err, common.ErrAccessToken):
+			// Retryable, so just log and retry
+			slog.Warn("Access token invalid, retrying", "error", err)
```

Comment:
It's not retrying though. It should but it's not. In the Salesforce connector I was (at one point) doing this, but it turned out that the retry library I was using had an incompatible license, so I ripped it out. Bits like this are left over from that, but you probably don't want to carry that over to the hubspot connector.

Also there's an entire conversation to be had about where retry logic should happen (in the connector or in the backend).

Reply:
Sure - keeping it as a todo item